DESIGN_PRINCIPLES.md

```diff
@@ -0,0 +1,24 @@
+# Concourse Design Principles
+
+Concourse's goal is to solve automation, once and for all, without becoming part of the problem.
+
+
+## Expressive by being precise
+
+Concourse should provide concepts that build a strong mental model for the user's project that remains intuitive as their automation grows.
+
+Concepts should precisely outline their motivation and intended workflows. Friction and complexity resulting from the imprecise application of a concept should be a cue to introduce new ideas. ([Example](https://blog.concourse-ci.org/reinventing-resource-types/))
+
+
+## Versatile by being universal
+
+Concourse should be able to do a lot with a little. New concepts should only be introduced if their intended workflow cannot be precisely expressed in terms of existing concepts.
+
+Concepts should not be highly specialized for one domain or introduce tight coupling to specific technologies. Users should be able to relate to every concept, and their automation should be insulated from the constant churn of the tech industry.
+
+
+## Safe by being destructible
+
+Concourse should prevent [anti-patterns](https://github.com/concourse/concourse/wiki/Anti-Patterns) and the accumulation of technical debt. Concourse's concepts should make good practices feel intuitive and bad practices feel uncomfortable.
+
+Automation should portable and reproducible, able to be recovered with minimal disruption to the path to production. All mission-critical state should be kept external to Concourse, leaving Concourse to just be a connector-of-dots.
```

Comment:
"Automation should **be** portable and reproducible, **and recovery should be possible** with minimal disruption to the path to production. All mission-critical state should be kept external to Concourse, leaving Concourse to just be a connector-of-dots."

Should we consider 'choreographer' or 'orchestration' instead of 'connector-of-dots'? 🤔

Reply:
lol yes probably

Reply:
Though there's also some aspect of statelessness I'd like to convey there - the dots metaphor comes from [this review comment](https://github.com/concourse/rfcs/pull/36#pullrequestreview-336557490) which made a point that Concourse itself is not a 'dot'.

Reply:
pushed another edition! lemme know what you think

Reply:
looks like you're still missing the corrections to the first sentence outline in https://github.com/concourse/rfcs/pull/97#discussion_r596038405 (esp. adding `be` to `Automation should ... portable`)

Reply:
woops, didn't even notice the bold text there - fixing!